### ISSUE TITLE
Thunderbird hangs when opening messages from search

### ISSUE THREAD

**Issue by trife** [thunderbird-conversations/thunderbird-conversations#1833]
* **Thunderbird version:** 91.9
* **Thunderbird Conversation version:** 4.0.13
* **Operating System:** Windows 10

**What did you do?** Search for a message, open search result in new tab which displays the dual pane message view

**What did you expect to happen?** Thread would display in left pane; selected message would load and display in right pane

**What actually happened?** Right pane does not load and Thunderbird CPU/RAM usage spikes. I generally have to kill the process. I only noticed this behavior today.

Potential bypass: is there any way to open messages directly to conversation view and bypass the dual pane search results view?

**Comment by alexbeldan** [comment]
Exact same issue here.

**Comment by darthoctopus** [comment]
I can confirm this is happening on Linux as well (Arch Linux), on two different machines with version 91.8.0 and 91.6.0 of Thunderbird. With debug output enabled, I get the following output on stdout after selecting a search result:

```
console.debug: "Initializing message pane view."
console.log: "Displaying" [{id:2, headerMessageId:"aef6a45c0e01e523aee2dda7a.6bf9809703.20220502200315.96d45339a9.b4fe4e66@mail169.sea81.mcsv.net"}, <long list of jsons>, {id:58417, headerMessageId:"010001808bfd2f6c-cb7892ad-9445-47bc-a167-85d448339f13-000000@email.amazonses.com"}, {id:58418, headerMessageId:"010001808c1d9fc8-258ea690-f5f8-48b0-a041-b2b8ed50d917-000000@email.amazonses.com"}]
console.error: "kScrollSelected && didn't find the selected message"
```

If I were to speculate, it seems that something is iterating through every email I've ever downloaded (>50k in this case) and erroring out.

**Comment by Standard8** [comment]
Thank you for the reports. I can reproduce it locally, and started looking at a fix. I've found the main problem - there's a small function in Thunderbird that the add-on uses that doesn't cover all the cases it could do, as a result we try to get the selected messages for the main tab rather than the search tab. As the main tab is effectively not showing anything at the time, that's an empty list - which we then do a search for to find all related, which seems to return all messages...

I've fixed the bit around getting the selected messages for the tab, and prevented trying to ever load an empty list. I still have to fix a bug that switching between search and normal tabs doesn't update the message pane. I think I know what I want to do for that, but I've run out of time for now.

I'll try and get to that either later today or tomorrow.

**Comment by maximelehericy** [comment]
I face a similar behavior of Thunderbird (loading, and Thunderbird CPU/RAM usage spikes) since today.
I filter on tags, and when I click on one of the results, it seems that the addon links together a lot of emails that have nothing in common (not the same conversation), and that triggers a resources spike.
Downgrading in 4.0.12 solves the issue.

**Comment by Standard8** [comment]
> I face a similar behavior of Thunderbird (loading, and Thunderbird CPU/RAM usage spikes) since today. I filter on tags, and when I click on one of the results, it seems that the addon links together a lot of emails that have nothing in common (not the same conversation), and that triggers a resources spike. Downgrading in 4.0.12 solves the issue.

Please file a separate issue for this, as it seems like it is not quite the same. Please also include exact steps to repeat, like exactly which functions/buttons you are using to filter - at a quick glance I can't reproduce it.

**Comment by k11i** [comment]
Don't know for sure if this is the same or an unrelated problem:

Since today Messages are not shown in the message pane (well, the one on the right, where the content of the mails is show).  I get the string "Loading ...." in the top bar of the pane (in the bar where the icons for delete conversation, archive conversation, show in tab and so on are).

If I deactivate Thunderbird Conversations, the message content is shown, but that is naturally not what I want. ;-)

I got:
- Debian Linux
- Thunderbird 91.8.0
- Thunderbird Conversations 
Version 4.0.13
Last Updated 4 May 2022

I cannot currently identify any debug output in the console log, I tried to activate debugging and "Inspect" the extention, but the console stays empty.

If you could tell me how to find and supply additional information, I will try to do so.

**Comment by Standard8** [comment]
As requested in my previous comment, please file separate issues for items that are not directly related to this one.

You can gain more debugging information by going to the three bar menu -> Tools -> Developer Tools -> Error Console and enabling "Show Content Messages" from the cog in the top right.

**Comment by Standard8** [comment]
I've made a few changes which definitely will fix the originally reported issue, and should fix some of the other issues as well.

I've posted it for review, I'm currently waiting on the Thunderbird add-ons team for that to be approved.

**Comment by Standard8** [comment]
4.0.14 is out now which will hopefully help. If it is still an issue please include exact details of what you are doing to trigger it, and include logs from the error console.

**Comment by trife** [comment]
Original issue solved for me so closing. My alternative question was never answered, though: is there any way to open messages directly to conversation view and bypass the dual pane search results view? I've always viewed that dual pane results view as an unnecessary step but didn't know if it was just a Thunderbird quirk.

**Comment by Standard8** [comment]
> My alternative question was never answered, though: is there any way to open messages directly to conversation view and bypass the dual pane search results view? I've always viewed that dual pane results view as an unnecessary step but didn't know if it was just a Thunderbird quirk.

That's a Thunderbird quirk/issue.
